src/rpc/dial.rs

```diff
@@ -673,10 +683,15 @@ impl DialBuilder<WithCredentials> {
                 ))
             }
         };
-        let mdns_uri = webrtc::action_with_timeout(self.get_mdns_uri(), Duration::from_secs(5))
-            .await
-            .ok()
-            .flatten();
+        // NOTE(benjirewis): Use a duration of 1500ms for getting the mDNS URI. I've anecdotally
+        // seen times as great as 922ms to fetch a non-loopback mDNS URI. With an
+        // interface_with_loopback query interval of 250ms, 1500ms here should give us time for ~6
+        // queries.
```

Comment:
@cheukt / @stuqdog , I lowered both the `interface_with_loopback` query interval and the `get_mdns_uri` timeout  to allow for faster-completing mDNS discovery (@erh hopefully this fixes your issue with excessively long dials when mDNS is enabled but not used).